### PR TITLE
Valgrind.rst: Add note about old valgrind and -gdwarf-4 setting

### PR DESCRIPTION
**Pull request by cassella**
Valgrind.rst: Add note about old valgrind and -gdwarf-4 setting

### CODE REVIEW

**Comment by cassella** [comment]
@jabraham17 If I understood, using valgrind with an llvm backend is just not supported?

This PR leaves me with another thought: the error reports in the preceding lines suggest trying quickstart.  But `util/quickstart/setchplenv.bash` sets `CHPL_LLVM=system` for me:

```
. util/quickstart/setchplenv.bash 
Setting CHPL_HOME to /home/fortytwo/src/chapel
Updating PATH to include /home/fortytwo/src/chapel/bin/linux64-x86_64
                     and /home/fortytwo/src/chapel/util
Updating MANPATH to include /home/fortytwo/src/chapel/man
Setting CHPL_COMM to none
Setting CHPL_TASKS to fifo
Setting CHPL_MEM to cstdlib
Setting CHPL_HOST_MEM to cstdlib
Setting CHPL_GMP to none
Setting CHPL_RE2 to none
Setting CHPL_LLVM to system
```

so that doesn't seem like useful advice.

Maybe the error should direct the user to `doc/rst/developer/bestPractices/Valgrind.rst` instead?  And then that document should mention not using LLVM, which it doesn't.


FWIW, the error I get from valgrind when LLVM is in use is

```
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x1b
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x25
> ### unhandled dwarf2 abbrev form code 0x1b
> ==3996084== Valgrind: debuginfo reader: ensure_valid failed:
> ==3996084== Valgrind:   during call to ML_(img_get)
> ==3996084== Valgrind:   request for range [8566654, +4) exceeds
> ==3996084== Valgrind:   valid image size of 1676416 for image:
> ==3996084== Valgrind:   "/home/fortytwo/src/chapel/test/unstable/const-intent/constIntent-longer"
> ==3996084==
> ==3996084== Valgrind: debuginfo reader: Possibly corrupted debuginfo file.
> ==3996084== Valgrind: I can't recover.  Giving up.  Sorry.
> ==3996084==
```

**Comment by jabraham17** [comment]
I think I misspoke the other day, valgrind does work with LLVM. We use it in our nightly tests, see `util/cron/test-valgrind.bash`.

> FWIW, the error I get from valgrind when LLVM is in use is

That error looks similar to something I've seen in our internal task tracking issues. I believe its related to newer llvms using `-gdwarf-5` by default, but valgrind doesn't understand it. Try recompiling the runtime and your code with `-gdwarf-4`.

**Comment by cassella** [comment]
I changed the PR to modify the Valgrind bestPractices document to describe that workaround.

I decided since this is developer-facing documentation I didn't need to spell out "rebuild the runtime", but I could add that.

I spot checked the generated html.

File: doc/rst/developer/bestPractices/Valgrind.rst [hunk 1]
```diff
@@ -46,3 +46,43 @@ The above options are needed because not all third-party libraries support
   reasons, unless ``CHPL_RE2_VALGRIND_SUPPORT=true`` is set at build time
 - GASNet support for ``valgrind`` is experimental at this time -- see 
   https://github.com/chapel-lang/chapel/issues/8544 for the current status
+
+LLVM Compatibility
+------------------
+
+Some versions of LLVM generate debugging information that some version
+of valgrind can't understand.  This can manifest as an error like the following:
```
Comment (jabraham17) [inline doc/rst/developer/bestPractices/Valgrind.rst:54]: Small note about spelling, I think it should be "that some versions of valgrind"

**Review by jabraham17 — COMMENTED** [review]
Hi @cassella, this looks almost ready to go. Can you just fix this one small spelling note

File: doc/rst/developer/bestPractices/Valgrind.rst (hunk 1 above)
Comment (cassella) [reply doc/rst/developer/bestPractices/Valgrind.rst]: Hi @jabraham17 , oops, thanks.  Done.

**Review by jabraham17 — APPROVED** [review]

**Pull request merged**